src/Voron/Data/Tables/Table.cs

```diff
@@ -433,7 +436,7 @@ public void Delete(long id)
             var ptr = DirectReadRaw(id, out int size, out bool compressed);
 
             if (compressed)
-                _tx.CachedDecompressedBuffersByStorageId?.Remove(id);
+                ForgetAboutCompressed(id);
```

Comment:
Remove `ForgetAboutCompressed` compressed entirely and use `_tx.ForgetAbout` instead? What about other places where `.Remove` is used on that dictionary? Do we need to look into that?